runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -2432,8 +2432,14 @@ bool TR_EscapeAnalysis::checkAllNewsOnRHSInLoopWithAliasing(int32_t defIndex, TR
                      fieldName = underlyingArray->getSymbolReference()->getOwningMethod(comp())->staticName(underlyingArray->getSymbolReference()->getCPIndex(), fieldNameLen, comp()->trMemory());
                      }
 
-                  if (fieldName && (fieldNameLen > 0) &&
-                      !strncmp(fieldName, "java/lang/Integer$IntegerCache.cache", 36))
+                  if (fieldName && (fieldNameLen > 10) &&
+                        !strncmp("java/lang/", fieldName, 10) &&
+                          (!strcmp("Integer$IntegerCache.cache", &fieldName[10]) ||
```

Comment:
I think we need to use `strncmp` here because names are not guaranteed to be null terminated.

Reply:
Otherwise - I think this looks good.

Reply:
OK.  Now I see `staticName` yields the name and type signature, separated by a space.  I'll fix that.

Reply:
Corrected that blunder in commit 4bf7ad3bb06a85e54e7d8562d875ffc474c2f8fd